### DISCUSSION TITLE
How can I get as single record based off its `name` field?

### DISCUSSION THREAD

**Discussion by ItsTerm1n4l** [pocketbase/pocketbase#4107] [Q&A] [answered]
Is there any way to retrieve a record by name instead of id?
like this:
```
const record1 = await pb.collection('posts').getOne('RECORD_ID');
```
but by record name.
const record1 = await pb.collection('posts').getOne('Name field');

**Comment by ganigeorgiev** [comment, marked as answer]
You can use the `getFirstListItem()` SDK helper:

```js
const record = await pb.collection('posts').getFirstListItem('name="something"');
```

**Reply by ItsTerm1n4l** [reply]
how do I replace `name="something"` with a variable?

**Reply by ItsTerm1n4l** [reply]
i tried `'name=var'` but i get errors and my site crashes.

**Reply by ganigeorgiev** [reply]
You can use [JS template literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) or the `pb.filter()` helper:

```js
let yourVar = "something";

// with string interpolation/template literal
const record = await pb.collection('posts').getFirstListItem(`name="${yourVar}"`);

// or with pb.filter
// https://github.com/pocketbase/js-sdk?tab=readme-ov-file#binding-filter-parameters
const record = await pb.collection('posts').getFirstListItem(pb.filter("name={:name}", {
    name: yourVar
}));
```

**Reply by ItsTerm1n4l** [reply]
Thank you, I had tried JS template literal but it still didn't work, but looking at your answer I can see where I went wrong, I did 
`pb.collection('parts').getFirstListItem('name="${var}"')`
wheras I needed.
```pb.collection('parts').getFirstListItem(`name="${var}"`)```
